env0/resource_project.go

```diff
@@ -106,8 +114,44 @@ func resourceProjectDelete(ctx context.Context, d *schema.ResourceData, meta int
 	apiClient := meta.(client.ApiClientInterface)
 
 	id := d.Id()
-	err := apiClient.ProjectDelete(id)
-	if err != nil {
+
+	forceDestroy := d.Get("force_destroy").(bool)
+
+	if !forceDestroy {
+		// Wait up to a minute before returning an error.
+
+		ctx, cancel := context.WithTimeout(ctx, time.Minute*1)
+		defer cancel()
+
+		for {
+			envs, err := apiClient.ProjectEnvironments(id)
+			if err != nil {
+				return diag.Errorf("could not get project environments: %v", err)
+			}
+
+			// Filter out archived (destroyed) environments.
+			hasActiveEnvs := false
+
+			for _, env := range envs {
```

Comment:
Deleting an environment takes time. One option is to fix the delete environment to wait until it's deleted (my preferred option).

Another option would be to call delete on each "active" environment (I can then increase the timeout from 1 minute to 5 or 10).

(At the moment, there's no point in waiting 5-10 minutes if an environment remains active).

Let me know your thoughts.

Reply:
https://env0.slack.com/archives/C01QBBEGBEH/p1647344145698209